CHANGELOG.md

```diff
@@ -6,6 +6,7 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 
 #### Bug fixes and improvements
+- Introduced `RoutesSetCallback` parameter to `MapboxNavigation#setNavigationRoutes`, which is called after the routes are set or they have been declared invalid. If invalid, they will not be passed to `RoutesObserver#onRoutesChanged` [#5945](https://github.com/mapbox/mapbox-navigation-android/pull/5945).
```

Comment:
maybe?
```suggestion
- Introduced a `RoutesSetCallback` parameter to `MapboxNavigation#setNavigationRoutes`. The callback is called when navigation finished or failed processing given routes. [#5945](https://github.com/mapbox/mapbox-navigation-android/pull/5945)
-  `RoutesObserver#onRoutesChanged` won't be triggered if navigation fails processing new routes. [#5945](https://github.com/mapbox/mapbox-navigation-android/pull/5945)
```

Reply:
> > The callback is called when navigation finished
> 
> Why? It's called when routes are set, hence the name.

I just find "they have been declared invalid" a bit vague. I think "failed processing given routes" should be easier for users to understand. "set" just doesn't sound good with "failed processing given routes", but "finished" does.

It's a very subjective topic, so it would be nice to hear other opinions. WDYT?

Reply:
OK, first: "when navigation finished" seems odd. It's actually triggered when `setNavigationRoutes` method finished. I can also say "when routes processing finished" or something like that. I don't see how it's clearer than "routes are set", especially when we're talking about the method called `setNavigationRoutes` but I might lack depth in terminology knowledge.
 About "failed processing given routes": probably I agree, because for now the reason is their invalidity, maybe in the future there will be other reasons, your option covers it.
 Tried to create a grammatically valid sentence having taken these points into consideration. Please take a look.